set.go

```diff
@@ -0,0 +1,293 @@
+package ksuid
+
+import "encoding/binary"
+
+// CompressedSet is an immutable data type which stores a set of KSUIDs.
+type CompressedSet []byte
+
+// Iter returns an iterator that produces all KSUIDs in the set.
```

Comment:
What's the common use case of using the iterator? Wondering if most of the time folks would just want to get all the []KSUID in the list.

Reply:
If you generate a slice to hold the entire set, you kind of defeat the purpose of using a compressed data structure don't you?

Reply:
Ah I see, yep that makes sense